base/asyncevent.jl

```diff
@@ -127,8 +127,11 @@ function _trywait(t::Union{Timer, AsyncCondition})
         t isa Timer || Core.Intrinsics.atomic_fence(:acquire_release)
     else
         if !isopen(t)
```

Comment:
I think you need to make this an acquire/release pair also (set isopen to false with release ordering and make isopen use acquire ordering) to ensure the memory model matches the code order here. Otherwise though, this LGTM

Reply:
I don't quite understand what you are trying to say. Could you try it again with different words?

Reply:
Better still: make a code suggestion.